pkg/adsc/adsc.go

```diff
@@ -305,42 +315,47 @@ func (a *ADSC) tlsConfig() (*tls.Config, error) {
 	var err error
 	var certName string
 
-	// If we need MTLS - CertDir or Secrets provider is set.
-	if a.cfg.Secrets != nil {
-		tok, err := ioutil.ReadFile(a.cfg.JWTPath)
-		if err != nil {
-			log.Infof("Failed to get credential token in agent: %v", err)
-			tok = []byte("")
-		}
+	var getClientCertificate func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error)
 
-		certName = fmt.Sprintf("(generated from %s)", a.cfg.JWTPath)
-		key, err := a.cfg.Secrets.GenerateSecret(context.Background(), "agent",
-			cache.WorkloadKeyCertResourceName, string(tok))
-		if err != nil {
-			return nil, err
-		}
-		clientCert, err = tls.X509KeyPair(key.CertificateChain, key.PrivateKey)
-		if err != nil {
-			return nil, err
+	if a.cfg.SecretManager != nil {
+		getClientCertificate = func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
Can you just move this to a proper method (getClientCert)  ? not a big deal, can be done in separate PR

Reply:
ok